src/main/java/org/apache/commons/collections4/iterators/UnmodifiableIterator.java

```diff
@@ -79,4 +79,15 @@ public void remove() {
         throw new UnsupportedOperationException("remove() is not supported");
     }
 
+    /**
+     * Allows package scoped access to the wrapped iterator for a very specific IteratorChain usecase.
+     * @return the wrapped IteratorChain instance or null when wrapped iterator is not a IteratorChain
+     */
+    IteratorChain<? extends E> getPossibleUnderlyingIteratorChain() {
```

Comment:
Hello @JoergBudi 

Thank you for your PR.

I'm not a fan of special casing one type in an unwieldy-named method here. Instead, please rebase on git master and use `UnmodifiableIterator.unwrap()`.

TY!